server/protocols/webadmin/webadmin-mailbox/src/main/java/org/apache/james/webadmin/routes/ImportEmlRoutes.java

```diff
@@ -83,33 +83,33 @@ public void define(Service service) {
         @ApiResponse(code = HttpStatus.NO_CONTENT_204, message = "EML file successfully imported"),
         @ApiResponse(code = HttpStatus.BAD_REQUEST_400, message = "Invalid parameter")
     })
-    public HaltException importEmlFileToMailbox(Request request, Response response) throws MailboxException {
-
+    public HaltException importEmlFileToMailbox(Request request, Response response) throws Exception {
+        final byte[] fileSize = request.body().getBytes();
         MailboxSession session = getMailboxSessionFromEmlService();
         MessageManager mailbox = getMailboxFromEmlService(request);
 
-        try {
-            mailbox.appendMessage(MessageManager.AppendCommand.builder()
+        mailbox.appendMessage(MessageManager.AppendCommand.builder()
                 .recent()
                 .build(request.body()), session);
-        } catch (MailboxException e) {
+
+        if (fileSize.length > MAXIMUM_SIZE_IN_BYTE) {
             throw ErrorResponder.builder()
-                .statusCode(HttpStatus.BAD_REQUEST_400)
-                .type(ErrorResponder.ErrorType.INVALID_ARGUMENT)
-                .message("Invalid file extension.")
-                .haltError();
+            .statusCode(HttpStatus.INTERNAL_SERVER_ERROR_500)
```

Comment:
you basically never have to use INTERNAL_SERVER_ERROR_500, it's for unhandled problems.
Please learn HTTP codes to choose them wisely (HTTP 413 here)